HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,706 @@
+- The default value for automatically-created accounts will now be `-1`. This means, if an account is created automatically (by performing a token transfer to an alias that does not yet exist, see [HIP-583](https://hips.hedera.com/hip/hip-583)), then the automatically created account is configured with unlimited automatic token associations.
+
+### Crypto Transfers
+
+Normal crypto-transfers remain the same as today, with one difference. If Account A sends a token to Account B, and Account B is not associated with that token but has an available auto-association slot, then Account A will pay for that association and the first auto-renewal period’s rent, in addition to the typical transfer costs.
+
+![Crypto Transfer Transaction](../assets/hip-904/frictionless-airdrop-cryptoTransferTransaction.png)
+
+### Airdrops
+
+Airdrops are similar to crypto transfers, but differ in one fundamental way - when an airdrop is performed and the receiver does not have available or automatic association slots, rather than failing, the transfer will be kept in state as a *pending transfer*. To represent this new semantic, there will be a new `TokenAirdropTransaction` meant for use specifically by those attempting an airdrop, with the following semantics:
+
+![Token Airdrop Transaction](../assets/hip-904/frictionless-airdrop-tokenAirdropTransaction.png)
+
+> ⚠️ `TokenAirdropTransaction` will not support tokens with custom fallback fees that require the receiver to pay the fallback fee.
+
+
+When a sender airdrops a token, if the receiver’s account is already associated with the token, then it is effectively treated as a normal crypto transfer, except that an *airdrop fee* is assessed. This fee exists for all uses of `TokenAirdropTransaction` as a spam-deterrent fee.
+
+If the receiver’s account is **not** associated with the token, but has available auto-association slots, then one of the slots is taken (i.e. `used_auto_associations` is incremented) and the transaction payer will pay for the association fee. They also pay a fee representing the price of a full auto-renewal period’s rent. This is done to make sure the receiver has time to dissociate from the token before renewal takes place. For example, if the airdrop were to happen the minute before account renewal, the token would not included in renewal because it has already been pre-paid by the sender for one full renewal period. Thus, the receiver always has at least one full renewal period to dissociate from any tokens they do not want to own.
+
+If the receiver’s account is **not** associated with the token, and they have **no** free auto-association slots, then the airdrop transfer neither succeeds nor fails, but is held in state as a *pending transfer*.
+
+### Pending Transfers
```

Comment:
Is there any kind of rent fee for keeping a pending transfer alive indefinitely?  It's taking up space in the store.

(In the airdrop diagram there is a fee table slot for "future rents until claimed" - but that seems to be rent on the _token_ not the pending transfer as the same exact wording "future rents until claimed" appears on the right-hand-side for the token itself.)

Reply:
Good question.
I had a similar consideration and personally also think rent should factor in the pending state as a separate additional item.
Rent has yet to be designed fully but this will certainly be a consideration.

Reply:
The pending transfer will **definitely** have rent on it. This is why it was called about above in a previous comment.

Reply:
Hmm I agree in principle except rent hasn't been defined so i don't want to put in details wording that may be changed. Rent will have many components so i think just saying rent is assessed to the sender is enough for now

Reply:
A future HIP defining rent will explicitly call out all the contributors

Reply:
yes I understand we haven't defined rents yet, but the flow diagrams _already_ include table slots for "future rents".  I'm just suggesting here that there be a table slot for "future pending transfer rents", to make it clear, in the same way, that this is the intent for the future and that the sender pays.